core/view/src/parser/index.ts

```diff
@@ -44,6 +43,13 @@ export class Parser {
     onCreateASTNode: new SyncWaterfallHook<
       [Node.Node | undefined | null, object]
     >(),
+
+    determineNodeType: new SyncBailHook<[object | string], NodeType>(),
+
+    parseNode: new SyncBailHook<
+      [object | null, Node.ChildrenTypes, ParseObjectOptions, NodeType | null],
```

Comment:
For hook types with longer definitions, you can name the arguments to keep track a bit easier:

```suggestion
      [
        obj: object | null,
        nodeType: Node.ChildrenTypes, 
        parseOptions: ParseObjectOptions, 
        determinedNodeType: NodeType | null
        ],
```